pkg/operator/certrotation/signer.go

```diff
@@ -75,7 +75,7 @@ func (c RotatedSigningCASecret) EnsureSigningCertKeyPair(ctx context.Context) (*
 	// apply necessary metadata (possibly via delete+recreate) if secret exists
 	// this is done before content update to prevent unexpected rollouts
 	if ensureMetadataUpdate(signingCertKeyPairSecret, c.Owner, c.AdditionalAnnotations) && ensureSecretTLSTypeSet(signingCertKeyPairSecret) {
-		actualSigningCertKeyPairSecret, _, err := resourceapply.ApplySecret(ctx, c.Client, c.EventRecorder, signingCertKeyPairSecret)
+		actualSigningCertKeyPairSecret, _, err := resourceapply.ApplySecretDoNotUse(ctx, c.Client, c.EventRecorder, signingCertKeyPairSecret)
```

Comment:
we should expose an option on `EnsureSigningCertKeyPair` that will force the usage of `ApplySecretDoNotUse`. Otherwise components that pull the new version will be broken.

Reply:
yes, that makes sense, this way we have the client opt-in for the new behavior we are adding, so just revendoing library-go does not automatically changes the client to use the new behavior